### DISCUSSION TITLE
Record/log node_modules locations in store for easy removal

### DISCUSSION THREAD

**Discussion by stevenlele** [pnpm/pnpm#15994] [Ideas]
- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

**Describe the user story**

When trying to remove all packages in store, you need to manually remove all `node_modules` installations by pnpm, then run `pnpm store prune`. Even if you managed to remove them from all projects you remember having in your disk, there might be some secret installations you don't know. Windows users can use tools like `Everything` to search for `node_modules` or `.pnpm` folders, but macOS users don't have such tool. For example, I had to use `ls -l` then `find ~ -samefile` on files in the store to find that `~/Library/Caches/typescript` had hard linked files from the store.

**Describe the solution you'd like**

Whenever `pnpm install` is triggered in any way or from any source, log the `node_modules` location to the corresponding store. It could be a simple text file for user to check upon later.

**Describe the drawbacks of your solution**

There might be some privacy consideration but I think it's generally fine.

**Describe alternatives you've considered**

Maybe a command to automatically find existing installations (e.g. `find ~ -samefile` store file on Unix-like systems), but that would be much more complicated.

**Comment by zkochan** [comment]
I don't have objections. But when this will be implemented, keep in mind that there may be several processes running install and it should fail if multiple pnpm instances will try to write to the file at the same time. We shouldn't use fs locking. It is an edge case, so it is OK if one of the processes will fail to update the list.

I wonder if we need to also store the locations of all the stores on the computer. There could be multiple.

**Comment by altaywtf** [comment]
I'm working on a fix for this.

Written by an agent (Claude Code, Opus 5.5).

**Comment by altaywtf** [comment]
A fix is ready for review in https://github.com/pnpm/pnpm/pull/15862. CI is green.

Written by an agent (Claude Code, Opus 5.5).
